tb_plugin/test/test_compare_with_aurograd.py

```diff
@@ -0,0 +1,176 @@
+        s = str(evt.device_type)
+        postfix = s[s.index('.') + 1:]
+        evt_type = postfix_to_type[postfix]
+        for prefix in others_prefix:
+            if evt.key.startswith(prefix):
+                evt_type = "Other"
+                break
+        return evt_type
+
+    result_dict = dict()
+    result_dict[worker_name + "#operator"] = list()
+    result_dict[worker_name + "#kernel"] = list()
+    for avg in avgs:
+        evt_type = get_type(avg)
+        if evt_type == "operator":
+            line = [avg.key, int(avg.count)]
+            if is_gpu:
+                line.extend([int(avg.self_cuda_time_total), int(avg.cuda_time_total)])
+            line.extend([int(avg.self_cpu_time_total), int(avg.cpu_time_total)])
+            result_dict[worker_name + "#operator"].append(line)
+        elif is_gpu and evt_type == "kernel":
+            line = [avg.key, int(avg.count), int(avg.self_cuda_time_total)]
+            result_dict[worker_name + "#kernel"].append(line)
+    return result_dict
+
+def get_plugin_result(run):
+    result_dict =  dict()
+    for worker_name, profile in run.profiles.items():
+        if not profile.operation_table_by_name is None:
+            rows = profile.operation_table_by_name["data"]["rows"]
+            result_dict[worker_name + "#operator"] = rows
+            if not profile.kernel_table is None:
+                rows = profile.kernel_table["data"]["rows"]
+                result_dict[worker_name + "#kernel"] = list()
+                for row in rows:
+                    result_dict[worker_name + "#kernel"].append(row[:3])
+    return result_dict
+
+def get_train_func():
+    model = models.resnet50(pretrained=True)
+    model.cuda()
+    cudnn.benchmark = True
+
+    transform = T.Compose([T.Resize(256), T.CenterCrop(224), T.ToTensor()])
+    trainset = torchvision.datasets.CIFAR10(root='./data', train=True,
+                                            download=True, transform=transform)
+    trainloader = torch.utils.data.DataLoader(trainset, batch_size=32,
+                                              shuffle=True, num_workers=0)
+
+    criterion = nn.CrossEntropyLoss().cuda()
+    optimizer = torch.optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
+    device = torch.device("cuda:0")
+    model.train()
+
+    def train(train_step, prof=None):
+        for step, data in enumerate(trainloader, 0):
+            print("step:{}".format(step))
+            inputs, labels = data[0].to(device=device), data[1].to(device=device)
+
+            outputs = model(inputs)
+            loss = criterion(outputs, labels)
+
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            if prof is not None:
+                prof.step()
+            if step >= train_step:
+                break
+    return train
+
+def get_output_fn(dir_name, profilers_dict):
+    def output_fn(p):
+        profilers_dict["worker{}".format(p.step_num)] = p.profiler
```

Comment:
Consider adding a comment: "In current torch.profiler.profile, at beginning of each span, a new p.profiler will be created. So the same p.profiler will not be shared among different spans."

Reply:
done